File: SaturdayQuizWeb/Scripts/Quiz.ts
import Question from "./Question.js";

export default class Quiz {
    private readonly _date: Date;
    private readonly _questions: Question[];

    constructor(jsonObject) {
        this._date = new Date(jsonObject.date);
        this._questions = jsonObject.questions.map(q => new Question(q));
    }

    get date(): Date {
        return this._date;
    }

    get questions(): Question[] {
        return this._questions;
    }
}

File: SaturdayQuizWeb/Models/Quiz.cs
namespace SaturdayQuizWeb.Models;

public sealed record Quiz
{
    public required string Id { get; init; } = string.Empty;

    public required DateTime Date { get; init; }

    public required string Title { get; init; } = string.Empty;

    public required IEnumerable<QuestionModel> Questions { get; init; } = new List<QuestionModel>();
}
